torch/nn/modules/activation.py

```diff
@@ -735,6 +736,12 @@ def _reset_parameters(self):
         if self.bias_v is not None:
             xavier_normal_(self.bias_v)
 
+    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
+                              missing_keys, unexpected_keys, error_msgs):
+        super(MultiheadAttention, self)._load_from_state_dict(
```

Comment:
Isn't this... just a useless override? If you want to do BC support, you should do some version comparison and state_dict manipulations. See `BatchNorm*` for an example.

Reply:
Because `_qkv_same_embed_dim` added to PyTorch v.1.2.0 is a non-learnable parameter, we don't need to add any other parameters here. This func will load the learnable parameters (`in_proj_weight`, `in_proj_bias` etc) from an old model to the new model. See my test above (I had an old model trained by PyTorch v.1.1.0 and able to call `load_state_dict` for inference).

Reply:
`_qkv_same_embed_dim` was not a parameter, so it is not in the state dict. I think here you are calling `MultiheadAttention`'s constructor, which is adding `_qkv_same_embed_dim` for you. `_load_from_state_dict`  doesn't do anything automatically and here is just a no-op override. The problem would be loading a model like this

In 1.1:
```python
torch.save(nn.MultiheadAttention(...), 'my_old_model.pt')
```

And loading it in 1.3:

```python
torch.load('my_old_model.pt')
```

The constructor does not get called by `pickle` in `torch.load`, so this is where you would run into a BC-incompatibility issue.

Reply:
@apaszke Yes, using [setattr](https://docs.python.org/3/library/pickle.html#object.__setstate__) makes sense. So I suppose we should overwrite setstate and set ```_qkv_same_embed_dim``` to the default value before continuing the pickle process by calling into the superclass? That'll make sure that, if the pickle state doesn't contain that attribute, it'll still be defined.

Reply:
I am really confused here. Why does this override do anything? `_qkv_same_embed_dim` is just a regular `int` attribute, and is not even related to state dict. Of course loading an old state dict works because the old and the new state dicts have the exact same set of keys, and this attribute is set already in `__init__` of the module. Am I missing anything here...?

Reply:
Yes, let's confirm this whole PR is even necessary, i.e. we currently don't have backwards compatibility with models trained with 1.1